### ISSUE TITLE
How to enable bounding box argument for `keras.layers.Random*`

### ISSUE THREAD

**Issue by innat** [keras-team/keras-cv#295]
This includes:

```python
- tf.keras.layers.RandomFlip
- tf.keras.layers.RandomCrop
- tf.keras.layers.RandomTranslation
- tf.keras.layers.RandomRotation
- tf.keras.layers.RandomZoom
- tf.keras.layers.RandomWidth
- tf.keras.layers.RandomHeight
```

**Comment by LukeWood** [comment]
These will be nontrivial:

- RandomFlip: flip the bounding box over the axis it is flipped
- RandomCrop: shift the bounding box offsets to match the cropped result
- RandomTranslation: translate bounding box the same amount as image
- RandomRotation: convert bounding box to segmentation map, rotate segmentation map same amount as image, convert back to bounding box?  Maybe we can come up with a better solution
- RandomZoom/Width/Height: scale bounding box scale bounding box

**Comment by sebastian-sz** [comment]
I'm not sure if it's possible to use these layers "as is" with bounding boxes (or segmentation maps).

For example, one would have to determine whether the augmentation actually took place and we don't know this unless we look at image pixels. Also, in case of more complicated layers (RandomZoom) to transform boxes one would need to know "how much" is each of the images zoomed. We also don't know this when applying the layer to the image.

I think one would have to to create a keras-cv wrapper's around these layers, that handle bounding box and maybe even segmentation maps if possible. Thoughts?

**Comment by innat** [comment]
All these layers are assigned to @divyashreepathihalli. I think they have a plan.

**Comment by LukeWood** [comment]
Yep - closed.
